engine/test/test_path.cpp

```diff
@@ -0,0 +1,111 @@
+/* Copyright (C) 2003-2017 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include "gtest/gtest.h"
+
+#include "prefix.h"
+#include "util.h"
+
+struct TestData
+{
+    const char *path;
+    const char *dir;
+    const char *base;
+};
+
+static const TestData kUnixTestData[] =
+{
+    { "/", "/", "" },
+    { "/foo", "/", "foo" },
+    { "/foo/", "/foo", "" },
+    { "/foo/bar", "/foo", "bar" },
+    { "/foo/bar/", "/foo/bar", "" },
+    { "", "", "" },
+    { "foo", "", "foo" },
+    { "foo/", "foo", "" },
+    { "foo/bar", "foo", "bar" },
+    { "foo/bar/", "foo/bar", "" },
```

Comment:
In the first case it should be correct - it will return /foo//, bar - the second case I'll have to check. Do you think that split should do any normalization? It's a purely string based operation at the moment.